File: 351003/Zarenok/demo/src/main/java/com/example/demo/service/MarkService.java
package com.example.demo.service;

import com.example.demo.dto.requests.MarkRequestTo;
import com.example.demo.dto.responses.MarkResponseTo;
import com.example.demo.model.Mark;
import com.example.demo.repository.MarkRepository;
import org.springframework.stereotype.Service;

import java.time.ZonedDateTime;
import java.util.List;

@Service
public class MarkService {
    private final MarkRepository repository;
    private final EntityMapper mapper;

    public MarkService(MarkRepository repository, EntityMapper mapper) {
        this.repository = repository;
        this.mapper = mapper;
    }


    public MarkResponseTo create(MarkRequestTo dto) {
        Mark mark = mapper.toEntity(dto);
        Mark saved = repository.save(mark);
        return mapper.toMarkResponse(saved);
    }

    public List<MarkResponseTo> findAll() {
        List<Mark> list = repository.findAll();
        return mapper.toMarkResponseList(list);
    }

    public MarkResponseTo findById(Long id) {
        Mark mark = repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Mark not found: " + id));
        return mapper.toMarkResponse(mark);
    }

    public MarkResponseTo update(Long id, MarkRequestTo dto) {
        Mark existing = repository.findById(id)
                .orElseThrow(() -> new RuntimeException("Mark not found: " + id));

        mapper.updateMark(dto, existing);
        Mark updated = repository.save(existing);
        return mapper.toMarkResponse(updated);
    }

    public void delete(Long id) {
        if (!repository.existsById(id)) {
            throw new RuntimeException("Mark not found: " + id);
        }
        repository.deleteById(id);
    }
}

File: 351003/Pulish/app/services/mark_service.py
from app.dto.mark import MarkRequestTo, MarkResponseTo
from app.models.mark import Mark
from app.repository.inmemory import InMemoryRepository
from app.core.exceptions import NotFoundException


class MarkService:
    def __init__(self):
        self.repo = InMemoryRepository()

    def create(self, dto: MarkRequestTo) -> MarkResponseTo:
        mark = Mark(id=0, name=dto.name)
        saved = self.repo.create(mark)
        return self._to_response(saved)

    def find_all(self):
        return [self._to_response(m) for m in self.repo.find_all()]

    def find_by_id(self, id: int):
        mark = self.repo.find_by_id(id)
        if not mark:
            raise NotFoundException("Mark not found", 40403)
        return self._to_response(mark)

    def update(self, id: int, dto: MarkRequestTo):
        mark = self.repo.find_by_id(id)
        if not mark:
            raise NotFoundException("Mark not found", 40403)
        mark.name = dto.name
        return self._to_response(mark)

    def delete(self, id: int):
        mark = self.repo.find_by_id(id)
        if not mark:
            raise NotFoundException("Mark not found", 40401)
        self.repo.delete(id)

    def _to_response(self, mark: Mark) -> MarkResponseTo:
        return MarkResponseTo(id=mark.id, name=mark.name)
